flask_jwt_extended/jwt_manager.py

```diff
@@ -373,21 +399,57 @@ def claims_verification_failed_loader(self, callback):
         self._claims_verification_failed_callback = callback
         return callback
 
+    def additonal_claims_verification_loader(self, callback):
+        """
+        This decorator sets the callback function that will be called when
+        a protected endpoint is accessed, and will check if the custom claims
+        in the JWT are valid. By default, this callback is not used. The
+        error returned if the claims are invalid can be controlled via the
+        :meth:`~flask_jwt_extended.JWTManager.additonal_claims_verification_loader`
+        decorator.
+
+        This callback must be a function that takes one argument, which is the
+        custom claims (python dict) present in the JWT, and returns `True` if the
+        claims are valid, or `False` otherwise.
+        """
+
+        self._additonal_claims_verification_callback = callback
+        return callback
+
+    def additonal_claims_verification_failed_loader(self, callback):
+        """
+        This decorator sets the callback function that will be called if
+        the :meth:`~flask_jwt_extended.JWTManager.additonal_claims_verification_loader`
+        callback returns False, indicating that the user claims are not valid.
+        The default implementation will return a 400 status code with the JSON:
+
+        {"msg": "Additional claims verification failed"}
+
+        This callback must be a function that takes no arguments, and returns
+        a Flask response.
+        """
+        self._additonal_claims_verification_failed_callback = callback
+        return callback
+
+
     def _create_refresh_token(self, identity, expires_delta=None):
         if expires_delta is None:
             expires_delta = config.refresh_expires
 
         if config.user_claims_in_refresh_token:
             user_claims = self._user_claims_callback(identity)
+            additional_claims = self._additonal_claims_callback(identity)
         else:
             user_claims = None
+            additional_claims = None
```

Comment:
The additional claims (`aud` for example) should probably be in the refresh token regardless of the `user_claims_in_refresh_token` option.